course_catalog/tasks.py

```diff
@@ -98,7 +98,7 @@ def get_ocw_data():
             course_instance = None
         try:
             # fetch JSON contents for each course file in memory (slow)
-            for obj in raw_data_bucket.objects.filter(Prefix=course_prefix):
+            for obj in raw_data_bucket.objects.filter(Prefix=course_prefix + "/"):
```

Comment:
You might potentially end up filtering by a `Prefix` with two trailing slashes here, if `course_prefix` is assigned to `key_pieces[0] + "/"` a few lines up.

Maybe change to `(Prefix=course_prefix)`, and include the trailing slash in the `course_prefix` value if necessary?

Reply:
Done